fixpatches/src/commonMain/kotlin/com/saveourtool/sarifutils/adapter/SarifFixAdapter.kt

```diff
@@ -381,4 +456,28 @@ class SarifFixAdapter(
 
     private fun Replacement.prettyString(): String = "(startLine: ${this.deletedRegion.startLine}, endLine: ${this.deletedRegion.endLine}, " +
             "startColumn: ${this.deletedRegion.startColumn}, endColumn: ${this.deletedRegion.endColumn}, insertedContent: ${this.insertedContent})"
+
+    /**
+     * @return this path, relativized against the [test root][testRoot],
+     *   assuming this path is absolute and [test root][testRoot] is non-`null`.
+     */
+    private fun Path.relativeToTestRoot(): Path =
+            when (testRoot) {
+                null -> this
+                else -> relativeToSafe(testRoot)
+            }
+
+    private fun Path.relativeToFileSystemRoot(): Path =
+            when (val root = root) {
```

Comment:
Do we need a `val` for it?

Reply:
Yes, because `root` is a property, and smart cast doesn't work on it.

Reply:
Referenced from #40, marking as resolved for now.